### PR TITLE
espressif: update hal to latest v5.1.4

### PR DESCRIPTION
**Pull request by sylvioalves**
This PR brings changes to allow updating hal_espressif to latest v5.1.4 repository.
It updates Wi-Fi and BLE libraries and improvements.

### CODE REVIEW

**Review by LucasTambor — DISMISSED** [review]

File: soc/espressif/esp32/mcuboot.ld [hunk 1]
```diff
@@ -65,9 +64,6 @@ SECTIONS
     *libkernel.a:device.*(.literal .text .literal.* .text.*)
     *libkernel.a:timeout.*(.literal .text .literal.* .text.*)
 
-    *(.literal.bootloader_mmap .text.bootloader_mmap)
```
Comment (marekmatej) [inline soc/espressif/esp32/mcuboot.ld:68]: aren't those used in the `bootloader_init()`

Comment (sylvioalves) [reply soc/espressif/esp32/mcuboot.ld:68]: This is not necessary in mcuboot booting context. Tested with all SoCs.

**Comment by marekmatej** [comment]
LGTM

**Review by wmrsouza — APPROVED** [review]

**Review by raffarost — APPROVED** [review]

**Comment by sylvioalves** [comment]
@jhedberg just a heads up here to request this PR to be merged before rc1 if within time windows. Thanks.

File: drivers/bluetooth/hci/Kconfig.esp32 [hunk 2]
```diff
@@ -16,4 +15,12 @@ choice ESP_BT_HEAP
 
 endchoice # ESP_BT_HEAP
 
-endif
+config ESP32_BT_CONTROLLER_STACK_SIZE
+	int "Bluetooth controller stack size"
+	default 4096
+
+config ESP32_BT_CONTROLLER_TASK_PRIO
```
Comment (uLipe) [inline drivers/bluetooth/hci/Kconfig.esp32:22]: I think this symbol should be ranged between the priorities level supported by zephyr.

An invalid setting here may result in a silent failure.

File: drivers/wifi/esp32/Kconfig.esp32 [hunk 3]
```diff
@@ -251,10 +291,62 @@ config ESP32_WIFI_RX_IRAM_OPT
 	  When this option is disabled, more than 17Kbytes of IRAM memory will be saved
 	  but Wi-Fi performance will be reduced.
 
+config ESP32_WIFI_MAX_THREAD_PRIORITY
```
Comment (uLipe) [inline drivers/wifi/esp32/Kconfig.esp32:294]: Same here about the priorities range.

File: soc/espressif/common/Kconfig [hunk 4]
```diff
@@ -32,7 +32,32 @@ config ESP_HEAP_RUNTIME
...
 	  This is a memory pool used in runtime to create a new heap memory.
 
+config ESP32_TIMER_TASK_STACK_SIZE
+	int "Stack size of the high resolution ESP Timer"
+	default 4096
+	help
+	  Set the stack size for the internal high resolution ESP Timer
+	  used in Wi-Fi and BLE peripherals.
+
+config ESP32_TIMER_TASK_PRIO
```
Comment (uLipe) [inline soc/espressif/common/Kconfig:42]: Same here.

**Review by uLipe — CHANGES_REQUESTED** [review]

**Comment by sylvioalves** [comment]
@uLipe I get your point but I do not agree with that. If you take a look at the majority of the drivers and subsystems in Zephyr, most of it define TASK/THREAD priority as done in this PR. This value should, of course, be within Zephyr's priority levels, which should be considered of "common knowledge". If you have a scenario or suggestion that really makes a difference, I shall change it.

https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr+THREAD_PRIO+path%3A**%2FKconfig&type=code

**Comment by uLipe** [comment]
Fair enough :)

**Review by uLipe — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**